src/aiida/cmdline/commands/cmd_code.py

```diff
@@ -241,28 +242,40 @@ def show(code):
 @options.SORT()
 @with_dbenv()
 def export(code, output_file, overwrite, sort):
-    """Export code to a yaml file."""
+    """Export code to a yaml file. If no output file is given, default name is created based on the code label."""
 
-    import yaml
+    other_args = {'sort': sort}
 
-    code_data = {}
+    fileformat = 'yaml'
 
-    for key in code.Model.model_fields.keys():
-        value = getattr(code, key).label if key == 'computer' else getattr(code, key)
+    if output_file is None:
+        output_file = pathlib.Path(f'{code.full_label}.{fileformat}')
 
-        # If the attribute is not set, for example ``with_mpi`` do not export it, because the YAML won't be valid for
-        # use in ``verdi code create`` since ``None`` is not a valid value on the CLI.
-        if value is not None:
-            code_data[key] = str(value)
+    elif not str(output_file).endswith('.yml') and not str(output_file).endswith('.yaml'):
+        output_file = pathlib.Path(f'{output_file!s}.{fileformat}')
```

Comment:
Still not quite sure why you insist on forcing a yml/yaml extension. An extension has no real effect on its content. So why enforce it if a user wants something else?